### PR TITLE
ADO-836: Support ADO Repos

### PR DESCRIPTION
**Pull request by vadim-galeev**
impt auth: "impt auth azure-repos" command added which adds Azure Repos credentials to auth file
impt test: Passing Azure Repos credentials (username, token) via environment variables added
impt test: "impt test azure-repos" command added
impt test: Creating and deleting Azure Repos credentials in a file added
impt test: Using Azure Repos credentials from a file during test deployment creation added
impt build: Using Azure Repos credentials from auth file or environment variables during deployment creation added
Command Manual and Changelog updated (other docs are not up-to-date in the forked impt)

Manual testing was done against the public EI cloud and private Eaton's cloud under Windows and Linux. Eaton's private ADO Repos was used during the testing.

### CODE REVIEW

File: bin/cmds/test/delete.js [hunk 1]
```diff
@@ -78,7 +88,7 @@ exports.builder = function (yargs) {
         [Options.ALL] : {
             demandOption : false,
             describe: Util.format('Includes --%s, --%s, --%s and --%s options.',
```
Comment (ekaterina-nbl) [inline bin/cmds/test/delete.js]: format constant should be modified accordingly

File: lib/Auth.js [hunk 2]
```diff
@@ -450,6 +450,34 @@ class Auth {
...
+            }).
+            then(password => {
+                authConfig.setAzureReposToken(newUser, password);
+                authConfig.setProfileType(newUser, UserInteractor.MESSAGES.AUTH_PROFILE_AZURE_REPOS);
+                return authConfig.save();
+            })
+            .then(() => UserInteractor.printResultWithStatus())
+            .catch(error => UserInteractor.processError(error));
+        } else {
+            console.log(options);
```
Comment (ekaterina-nbl) [inline lib/Auth.js]: Is it debug logging?

File: CommandsManual.md [hunk 3]
```diff
@@ -1655,6 +1683,7 @@ At the end of the command execution, information about the test configuration is
 | --test-file | -f | No | Yes | Test file name or pattern. All files located in the current directory and all its sub-directories whose names match the specified name or pattern are considered as files with test cases. This option may be repeated multiple times to specify multiple names and/or patterns. The values of the repeated option are combined by logical OR. Default: `"*.test.nut"` `"tests/**/*.test.nut"` |
 | --github-config | -i | No | Yes | A path to a GitHub credentials file. A relative or absolute path can be used. The specified file may not exist |
 | --bitbucket-srv-config | -b | No | Yes | A path to a Bitbucket Server credentials file. A relative or absolute path can be used. The specified file may not exist |
+| --azure-repos-config | -az | No | Yes | A path to an Azure Repos credentials file. A relative or absolute path can be used. The specified file may not exist |
```
Comment (ekaterina-nbl) [inline CommandsManual.md]: It's better to use single letter alias if possible.
For example `-r` looks like a good candidate. Please check if there are any conflicts with it.

File: CommandsManual.md [hunk 4]
```diff
@@ -302,6 +309,11 @@ The Auth file is of the form:
         "bitbucketSrvUser": "BITBUCKET_SERVER_USER",
         "bitbucketSrvToken": "BITBUCKET_SERVER_TOKEN",
         "profileType": "bitbucketSrv"
+    },
+    "AZURE_REPOS_USERNAME": {
+        "azureReposUser": "AZURE_REPOS_USER",
```
Comment (mm-gmbd) [inline CommandsManual.md]: The constant should be consistent between these two lines (both should be `AZURE_REPOS_USERNAME`).

**Review by mm-gmbd — APPROVED** [review]
One minor comment, but this looks good!

File: CommandsManual.md (hunk 4 above)
Comment (vadim-galeev) [reply CommandsManual.md]: Fixed it, thank you for the comment.

**Pull request merged**